providers/os/resources/networkinterface/routes.go

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package networkinterface
+
+import (
+	"io"
+	"net"
+	"sort"
+	"strings"
+
+	"github.com/cockroachdb/errors"
+	"go.mondoo.com/cnquery/v12/providers-sdk/v1/inventory"
+	"go.mondoo.com/cnquery/v12/providers/os/connection/shared"
+	"go.mondoo.com/cnquery/v12/providers/os/resources/powershell"
+)
+
+// netr is a helper struct to avoid passing the connection and platform
+// as function arguments.
+type netr struct {
+	connection shared.Connection
+	platform   *inventory.Platform
+}
+
+// Route represents a network route entry
+type Route struct {
+	Destination string
+	Gateway     string
+	Flags       []string
+	Interface   string
+	Platform    *inventory.Platform // Platform-specific flag handling
+}
+
+// Routes returns the network routes of the system.
+// This function dispatches to platform-specific implementations based on runtime detection.
+func Routes(conn shared.Connection, pf *inventory.Platform) ([]Route, error) {
+	n := &netr{conn, pf}
+
+	if pf.IsFamily(inventory.FAMILY_LINUX) {
+		return n.detectLinuxRoutes()
+	}
+	if pf.IsFamily(inventory.FAMILY_DARWIN) {
```

Comment:
This only works if it is darwin and local connection, it would fail on darwin with ssh. Ok if we do not support this case for now.